polkadot/xcm/xcm-executor/src/lib.rs

```diff
@@ -825,7 +826,29 @@ impl<Config: config::Config> XcmExecutor<Config> {
 				let old_holding = self.holding.clone();
 				let result = Config::TransactionalProcessor::process(|| {
 					let deposited = self.holding.saturating_take(assets);
+					let mut all_failed = true;
```

Comment:
nit:

I would use a "mirrored" boolean for clarity, smth like:
```suggestion
					let mut deposit_succeeded = false;
```

or even better, just don't mark anything and _always_ retry `failed_deposits` / `failed_ed_deposits` - they either work or throw some error which we will simply bubble up with `?`.